securityscanutils/commands/scan_version.go

```diff
@@ -0,0 +1,101 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/go-utils/cliutils"
+	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/go-utils/osutils/executils"
+	"github.com/solo-io/go-utils/securityscanutils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func ScanVersionCommand(ctx context.Context, rootOptions *RootOptions) *cobra.Command {
+	opts := &scanVersionOptions{
+		RootOptions: rootOptions,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "scan-version",
+		Short: "Run Trivy scans against images for a single version",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			results, err := doScanVersion(ctx, opts)
+			if err != nil {
+				return err
+			}
+			if results != nil {
+				contextutils.LoggerFrom(ctx).Infof(results.Error())
+			} else {
+				contextutils.LoggerFrom(ctx).Infof("No vulnerabilities found")
+			}
+			return nil
+		},
+	}
+
+	opts.addToFlags(cmd.Flags())
+
+	return cmd
+}
+
+type scanVersionOptions struct {
+	*RootOptions
+
+	imageRepository string
+	imageVersion    string
+	imageNames      []string
+}
+
+func (o *scanVersionOptions) addToFlags(flags *pflag.FlagSet) {
+	flags.StringVarP(&o.imageRepository, "image-repo", "r", securityscanutils.QuayRepository, "image repository to scan")
+	flags.StringVarP(&o.imageVersion, "version", "t", "", "version to scan")
+	flags.StringSliceVar(&o.imageNames, "images", []string{}, "comma separated list of images to scan")
+
+	cliutils.MustMarkFlagRequired(flags, "version")
+	cliutils.MustMarkFlagRequired(flags, "images")
+}
+
+type scanResults error
```

Comment:
While it's an improvement to log the result at info level, I would suggest that the results of the scan (whether vulnerabilities are found or not) could/should be printed rather than logged since they are the output of the program per se

Regardless, I don't see a compelling reason to represent the list of vulnerabilities as an error
We can use a `[]string` as a list of images and define a function to pretty-print it with whatever additional context may be useful
The result could be along the lines of 
```
Vulnerabilities found in the following images: 
- foo 
- bar
- baz
```
Instead of the much more cumbersome
```
{"level":"info","ts":"2022-12-01T11:13:19.827-0500","caller":"commands/scan_version.go:33","msg":"7 errors occurred:\n\t* vulnerabilities found for quay.io/solo-io/gloo:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/gloo-envoy-wrapper:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/discovery:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/ingress:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/sds:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/certgen:1.13.0-rc1\n\t* vulnerabilities found for quay.io/solo-io/access-logger:1.13.0-rc1\n\n"}
```
which is awkwardly long and repetitive and includes unheeded formatting directives

Reply:
Yeah great call, i'll update